.gitignore

```diff
@@ -8,3 +8,6 @@ node_modules
 .sw?
 *.jar
 *.xpi
+.idea/
+.vscode
+package-lock.json
```

Comment:
This is a file that actually _should_ be checked into the repository, so that everyone working on the project uses the same versions of the development dependencies.

Reply:
Was there a reason to not add it to git in the first place?

Reply:
It was not present in the repo so I thought it was intentional. Do you want me to commit the `package-lock.json` file in this pull request?

Reply:
I think @gijsk should ultimately make the call on that.

Reply:
I think the only reason it's not committed right now is that there wasn't one when we first made the library node-compatible (many years ago) and we've never remedied that. I think it's a good idea to commit it, but it should be a separate commit and PR. It probably shouldn't be added to gitignore in this PR.

Reply:
@gijsk I have undone the .gitignore changes, Just to make sure, Do you want me to make another issue and create a pull request adding `package-lock.json`?

Reply:
@ankushduacodes a new PR to add `package-lock.json` would be great, yes, thank you!